scripts/carthage.sh

```diff
@@ -1,14 +1,14 @@
-#!/usr/bin/env bash
+#!/usr/bin/env bash -e -o pipefail
 #
 # Needed to circumvent an issue with Carthage version < 0.37.0: https://github.com/Carthage/Carthage/issues/3019
 #
 # carthage.sh
 # Usage example: ./carthage.sh build --platform iOS
 
 VERSION="$(carthage version)"
-"$(dirname "$0")/versions.sh" "$VERSION" "0.37.0"
+comparison=$("$(dirname "$0")/versions.sh" "$VERSION" "0.37.0"; echo $?)
 
-if [ $? -ge 0 ]; then
+if [ "$comparison" -ge 0 ]; then
```

Comment:
It shouldn't be necessary to do things in this way given how the `versions.sh` script exits, at least in theory. Are you not seeing this in your testing?